kong/pdk/response.lua

```diff
@@ -472,6 +474,28 @@ local function new(self, major_version)
   end
 
 
+  ---
+  -- Redirect uri in the response sent to the client.
+  --
+  -- @function kong.response.redirect
+  -- @phases rewrite, access, header_filter, admin_api
+  -- @tparam string uri The name of the url to be redirect
+  -- @return Nothing; throws an error on invalid input.
+  -- @usage
+  -- kong.response.redirect("https://github.com/kong/kong")
+  function _RESPONSE.redirect(uri)
+    check_phase(rewrite_access_header)
+
+    local parsed_uri = url.parse(uri)
+    if not (parsed_uri and parsed_uri.host and parsed_uri.scheme) then
+      error("cannot parse '" .. uri .. "'", 2)
+    end
+
+    ngx.header[LOCATION_HEADER_NAME] = uri
+    return ngx.exit(301)
```

Comment:
`302` is in general a better default, but it should also be an argument.

We also need to construct body. Here is what Nginx does: http://lxr.nginx.org/source/src/http/ngx_http_special_response.c

We can add Kong twist to it.

This should also use `kong.response.exit` to make short-circuiting work correctly. I am also not sure about the validation there. Is it needed, is it correct?

Reply:
Easy use and understanding.

Reply:
Just a note here, with our current PDK, you can do this:

```lua
return kong.response.exit(302, "", { Location: "<uri>" })